cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -93,38 +94,74 @@ private HttpClient createHttpClient(String url) {
 
   @Override
   public void download(String url, Path target) {
+    List<String> httpProtocols = IdeVariables.HTTP_PROTOCOLS.get(context);
+    List<HttpClient.Version> protocolVersions = new ArrayList<>();
+
+    for (String httpProtocol : httpProtocols) {
+      switch (httpProtocol) {
+        case "HTTP_1_1":
+          protocolVersions.add(HttpClient.Version.HTTP_1_1);
+          this.context.debug("Added HTTP_1_1 protocol");
+          break;
+        case "HTTP_2":
+          protocolVersions.add(HttpClient.Version.HTTP_2);
+          this.context.debug("Added HTTP_2 protocol");
+          break;
+        default:
+          this.context.error("Unsupported protocol was set in HTTP_PROTOCOLS: " + httpProtocol);
+      }
+    }
+
+    Exception lastException = null;
+    for (HttpClient.Version version : protocolVersions) {
+      try {
+        this.context.debug("Trying to download: {} with HTTP protocol version: {}", url, version);
+        this.downloadWithHttpVersion(url, target, version);
+        return; // success
+      } catch (Exception ex) {
+        lastException = ex;
+      }
+    }
+
+    throw new IllegalStateException("Failed to download file from URL " + url + " to " + target, lastException);
+  }
+
+  private void downloadWithHttpVersion(String url, Path target, HttpClient.Version httpClientVersion) throws Exception {
 
     this.context.info("Trying to download {} from {}", target.getFileName(), url);
     mkdirs(target.getParent());
-    try {
-      if (this.context.isOffline()) {
-        throw CliOfflineException.ofDownloadViaUrl(url);
+
+    if (this.context.isOffline()) {
+      throw CliOfflineException.ofDownloadViaUrl(url);
+    }
+    if (url.startsWith("http")) {
+
+      HttpRequest request = HttpRequest.newBuilder()
+          .uri(URI.create(url))
+          .GET()
+          .version(httpClientVersion)
```

Comment:
If a new HTTP version gets supported in Java, we explicitly have to change our code.
Wouldn't it be better to do the following:
1. run without explicit version (latest version and most secure approach)
2. only if the user has defined fallback versions via property (that is empty by default) these will be tried in the specified order.

To make this method reusable, I would suggest providing `null` as version for 1. and add an if not null around `version` method for builder.

Reply:
I've adjusted it as suggested.